### PR TITLE
refactor: use client module for status checks in 02-client handlers

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Use client module directly for status checks in 02-client.

closes: #6911

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by gjermundgaraba — APPROVED** [review]
Nice 🚀

File: modules/core/02-client/keeper/client.go [hunk 1]
```diff
@@ -119,17 +119,17 @@ func (k *Keeper) UpgradeClient(
...
-		return errorsmod.Wrapf(types.ErrClientNotActive, "substitute client is not %s, status is %s", exported.Active, status)
+	if status := clientModule.Status(ctx, subjectClientID); status == exported.Active {
+		return errorsmod.Wrapf(types.ErrInvalidRecoveryClient, "cannot recover subject client (%s) with status %s", subjectClientID, status)
 	}
 
-	clientModule, err := k.Route(ctx, subjectClientID)
-	if err != nil {
-		return errorsmod.Wrap(types.ErrRouteNotFound, subjectClientID)
+	if status := clientModule.Status(ctx, substituteClientID); status != exported.Active {
+		return errorsmod.Wrapf(types.ErrClientNotActive, "cannot recover client using substitute client (%s) with status %s", subjectClientID, status)
```
Comment (crodriguezvega) [inline modules/core/02-client/keeper/client.go:132]:
```suggestion
		return errorsmod.Wrapf(types.ErrClientNotActive, "cannot recover client using substitute client (%s) with status %s", substituteClientID, status)
```

**Review by crodriguezvega — APPROVED** [review]
Left just a small fix in the logging. I will push the fix myself.

**Comment by colin-axner** [comment]
We should be removing `GetClientStatus` as well?

**Comment by colin-axner** [comment]
Checked and it looks like it might still be useful for exposing light client module functions from 02-client without requiring other modules (03-connection) to also all `.Route`

Going to assume @damiannolan implemented the changes with this in mind

**Review by colin-axner — APPROVED** [review]

**Pull request merged**